provider/provider.go

```diff
@@ -27,10 +27,15 @@ import (
 
 // Provider defines the interface DNS providers should implement.
 type Provider interface {
-	Records() ([]*endpoint.Endpoint, error)
+	Records(ctx context.Context) ([]*endpoint.Endpoint, error)
 	ApplyChanges(ctx context.Context, changes *plan.Changes) error
 }
 
+type EndpointModifyingProvider interface {
```

Comment:
Why does this needs to be a different interface? Do we have other options?

Also: this should have a comment.

Reply:
I just introduced this so that I wouldn't need to implement this as an empty method for all other providers and because I would consider this an "optional" interface of a provider. However, I wouldn't mind at all if you prefer to have this in the `Provider` interface.